alerting>docs>notification template example (#93330)
grafana/grafana: docs/sources/alerting/configure-notifications/template-notifications/create-notification-templates.md

   {{ template "email.message" . }}
   ```

## Group multiple alert instances into one email notification

To make alerts more concise, you can group multiple instances of a firing alert into a single email notification in a table format. This way, you avoid long, repetitive emails and make alerts easier to digest.

> Follow these steps to create a custom notification template that consolidates alert instances into a table:

Reviewer: ```suggestion
Follow these steps to create a custom notification template that consolidates alert instances into a table.
```

> 1. First, you need to modify the alert rule to include an annotation that will later be referenced in the notification template.

Reviewer: ```suggestion
1.Modify the alert rule to include an annotation that is referenced in the notification template later on.
```

> 1. Enter a name for the **custom annotation**: In this example, we'll use _ServerInfo_.

Reviewer: ```suggestion
1. Enter a name for the **custom annotation**: In this example, _ServerInfo_.
```

1. Enter the following code as the value for the annotation. It retrieves the server's instance name and a corresponding metric value, formatted as a table row:

   ```
   {{ index $labels "instance" }}{{- "\t" -}}{{ index $values "A"}}{{- "\n" -}}
   ```

   This line of code returns the labels and their values in the form of a table. Assuming $labels has `{"instance": "node1"}` and $values has `{"A": "123"}`, the output would be:

   ```
   node1    123
   ```

1. Create a notification template that references the _ServerInfo_ annotation.

   ```go
   {{ define "Table" }}
   {{- "\nHost\t\tValue\n" -}}
   {{ range .Alerts -}}
   {{ range .Annotations.SortedPairs -}}
   {{ if (eq .Name  "ServerInfo") -}}
   {{ .Value -}}
   {{- end }}
   {{- end }}
   {{- end }}
   {{ end }}
   ```

   The notification template outputs a list of server information from the "ServerInfo" annotation for each alert.

1. Apply the template you your contact point.
1. Navigate to your contact point in Grafana
1. In the **Message** reference the template by name (see **Optional Email settings** section):

   ```
   {{ template "Table" . }}
   ```

>    This will generate a neatly formatted table in the email, grouping information for all affected servers into a single notification.

Reviewer: ```suggestion
   This generates a neatly formatted table in the email, grouping information for all affected servers into a single notification.
```
